pkg/ingester/client/timeseries.go

```diff
@@ -245,7 +245,7 @@ func ReuseSlice(slice []PreallocTimeseries) {
 	for i := range slice {
 		ReuseTimeseries(slice[i].TimeSeries)
 	}
-	slicePool.Put(slice[:0])
+	slicePool.Put(slice[:0]) //nolint:staticcheck
```

Comment:
So this is the error from the linter.
```
pkg/ingester/client/timeseries.go:248:15: SA6002: argument should be pointer-like to avoid allocations (staticcheck)
	slicePool.Put(slice[:0])
	             ^
```

From my understanding of what we are trying to do here we want to reuse the slice but and reset it back to empty. What the linter would like is for everything that goes into a SlicePool to be a pointer. This is simply a slice of pointers so I guess my thought was that the overhead isn't a big deal. I will be completely honest here. I have not done much with SlicePools. But we can switch it to be a pointer as well. 

It'll just require more changes up the stack anywhere we pull `[]PreallocTimeseries` out of the slicePool.

Reply:
A slice itself is a pointer to an underlying array.  This is super minor.

Reply:
We cannot pool these as pointer-to-slice because the place we use them is in `WriteRequest` which is generated from Protobuf and we don't have an option to make it a pointer.

So the warning is correct: we generate 24 bytes of garbage every time a `PreallocTimeseries` is re-used.  But since the slices are far far larger, we come out ahead.

I think there should be a comment on `slicePool` to explain this.

Reply:
Comment added here 8b4be65